Added XLM onnx config

nandwalritik opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Added XLM OnnxConfig to make this model available for conversion.

@ChainYo 
## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
https://github.com/huggingface/transformers/issues/16308
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- ~~[ ] Did you write any new necessary tests?~~


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- albert, bert, xlm: @LysandreJik
- blenderbot, bart, marian, pegasus, encoderdecoder,  t5: @patrickvonplaten, @patil-suraj
- longformer, reformer, transfoxl, xlnet: @patrickvonplaten
- fsmt: @stas00
- funnel: @sgugger
- gpt2: @patrickvonplaten, @LysandreJik
- rag: @patrickvonplaten, @lhoestq
- tensorflow: @LysandreJik

Library:

- benchmarks: @patrickvonplaten
- deepspeed: @stas00
- ray/raytune: @richardliaw, @amogkam
- text generation: @patrickvonplaten
- tokenizers: @n1t0, @LysandreJik
- trainer: @sgugger
- pipelines: @LysandreJik

Documentation: @sgugger

HF projects:

- datasets: [different repo](https://github.com/huggingface/datasets)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Examples:

- maintained examples (not research project or legacy): @sgugger, @patil-suraj
- research_projects/bert-loses-patience: @JetRunner
- research_projects/distillation: @VictorSanh

 -->

nandwalritik committed:
Add onnx configuration for xlm

nandwalritik committed:
Add supported features for xlm

nandwalritik committed:
Add xlm to models exportable with onnx

nandwalritik committed:
Add xlm architecture to test file

nandwalritik committed:
Modify docs

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_17030). All of your documentation changes will be reflected on that endpoint.

nandwalritik committed:
Make code quality fixes

ChainYo commented:
Hello thanks for the PR it looks really clean!! 🤗

ChainYo commented on docs/source/en/index.mdx:
```diff
@@ -161,16 +160,13 @@ The library currently contains JAX, PyTorch and TensorFlow implementations, pret
 1. **[XLS-R](model_doc/xls_r)** (from Facebook AI) released with the paper [XLS-R: Self-supervised Cross-lingual Speech Representation Learning at Scale](https://arxiv.org/abs/2111.09296) by Arun Babu, Changhan Wang, Andros Tjandra, Kushal Lakhotia, Qiantong Xu, Naman Goyal, Kritika Singh, Patrick von Platen, Yatharth Saraf, Juan Pino, Alexei Baevski, Alexis Conneau, Michael Auli.
 1. **[YOSO](model_doc/yoso)** (from the University of Wisconsin - Madison) released with the paper [You Only Sample (Almost) Once: Linear Cost Self-Attention Via Bernoulli Sampling](https://arxiv.org/abs/2111.09714) by Zhanpeng Zeng, Yunyang Xiong, Sathya N. Ravi, Shailesh Acharya, Glenn Fung, Vikas Singh.
-
-
```
I saw multiple blank lines that have been removed, do you know why?

nandwalritik replied:
I just ran, the `make style`, `make fixup`, `make quality` commands for code reformatting, I think these blank lines got removed in running these commands as I didn't made any changes manually. If needed I can revert these changes which removed the blank lines.

ChainYo replied:
> I just ran, the `make style`, `make fixup`, `make quality` commands for code reformatting, 

Ok it seems to be fine. I usually only use `make fixup` because this command do everything (style + quality).

Let's see what Transformers maintainers think about this. @lewtun

nandwalritik replied:
I have reverted the formatting changes and rebased with `main`  branch.

lewtun reviewed (approved):
Thanks for this clean PR @nandwalritik 🔥 ! Apart from a small comment about the formatting changes, this LGTM :)

Could you please confirm that the slow tests pass with:

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "xlm"
```

Gently pinging @sgugger for final approval

sgugger commented on docs/source/en/index.mdx:
```diff
@@ -50,7 +50,6 @@ The library currently contains JAX, PyTorch and TensorFlow implementations, pret
 ### Supported models
 
 <!--This list is updated automatically from the README with _make fix-copies_. Do not update manually! -->
-
```
This change shouldn't be merged.

sgugger commented on docs/source/en/index.mdx:
```diff
@@ -161,16 +160,13 @@ The library currently contains JAX, PyTorch and TensorFlow implementations, pret
 Flax), PyTorch, and/or TensorFlow.
 
 <!--This table is updated automatically from the auto modules with _make fix-copies_. Do not update manually!-->
-
```
Please revert this as well.

sgugger commented on docs/source/en/serialization.mdx:
```diff
@@ -44,7 +44,6 @@ and are designed to be easily extendable to other architectures.
 Ready-made configurations include the following architectures:
 
 <!--This table is automatically generated by `make fix-copies`, do not fill manually!-->
-
```
Same here.

sgugger commented on docs/source/en/index.mdx:
```diff
@@ -275,5 +271,4 @@ Flax), PyTorch, and/or TensorFlow.
 |        XLMProphetNet        |       ✅       |       ❌       |       ✅        |         ❌         |      ❌      |
 |            XLNet            |       ✅       |       ✅       |       ✅        |         ✅         |      ❌      |
 |            YOSO             |       ❌       |       ❌       |       ✅        |         ❌         |      ❌      |
-
```
Same here.

sgugger commented on docs/source/en/serialization.mdx:
```diff
@@ -74,9 +73,9 @@ Ready-made configurations include the following architectures:
+- XLM
 - XLM-RoBERTa
 - XLM-RoBERTa-XL
-
```
And there.

nandwalritik replied:
Removed all these unwanted changes.

sgugger reviewed (commented):
Thanks for the PR! Good for me once the unwarranted changes are removed.

nandwalritik commented:
> Thanks for this clean PR @nandwalritik fire ! Apart from a small comment about the formatting changes, this LGTM :)
> 
> Could you please confirm that the slow tests pass with:
> 
> ```
> RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "xlm"
> ```
@lewtun The test cases are successfully passing on running `RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "xlm"`.

lewtun commented:
Thanks for checking the tests pass @nandwalritik ! Could you please rebase on `main` to account for a recent refactoring that was done to order the model names in `features.py` alphabetically?

lewtun commented on src/transformers/onnx/features.py:
```diff
@@ -345,6 +346,16 @@ class FeaturesManager:
             "token-classification",
             onnx_config_cls=RoFormerOnnxConfig,
         ),
+        "xlm": supported_features_mapping(
```
After rebasing, please put this entry in alphabetical order

nandwalritik replied:
Rebased with `main`, and changed  them according to alphabetical order, but few checks are failing due to changes from `Add YOLOS`. Can we ignore them or do I need to fix them?

lewtun replied:
For some reason your branch has picked up changes on the test files for `yolos`. Perhaps the rebase didn't work fully as expected? Once those `yolos` files are removed, the tests should pass :)

nandwalritik replied:
Thanks

nandwalritik commented:
> Hey @nandwalritik are you struggling with the commits or rebasing a branch ?

I just saw that there were merge conflicts, since `main` was updated, so I rebased again. Did I rebased incorrectly?
Steps which I followed to rebase:- 
* Fetch and merge upstream
* git pull origin main
* git checkout featureBranch
* git rebase main
And then I solved the merge conflicts manually wherever were required.

ChainYo commented:
Yes it seems that there is more than your commits attached to this PR

ChainYo commented:
Nice it seems to be better !

sgugger commented:
Thanks again for your contribution!
